lib/Doctrine/ORM/Proxy/ProxyFactory.php

```diff
@@ -37,6 +37,8 @@
  * @author Giorgio Sironi <piccoloprincipeazzurro@gmail.com>
  * @author Marco Pivetta  <ocramius@gmail.com>
  * @since 2.0
+ *
+ * @deprecated 3.0 This class is being removed from the ORM and won't have any replacement
```

Comment:
Well, actually, deprecating it in 2.7 looks wrong there, as it is impossible to avoid using it in 2.7 (as the migration to proxy-manager is not available in 2.7)

Reply:
That's completely fair. I'll address that in another PR, thanks

Reply:
Deprecating removed stuff with no replacement (which is exactly the case) is absolutely fine IMO.

Reply:
I completely misread the comments. What I thought was said was that the version is wrong in the deprecation notice (it's being deprecated in 2.7 not 3.0).

About it being deprecated now, it's completely fine since it's used internally and no error is being triggered.

Reply:
well, "deprecated" means you should stop using it because it will be removed. But here, there is no way to stop using it.